process/elasticproc/elasticProcessor.go

```diff
@@ -329,7 +329,23 @@ func (ei *elasticProcessor) RemoveTransactions(header coreData.HeaderHandler, bo
 		return err
 	}
 
-	return ei.removeIfHashesNotEmpty(elasticIndexer.LogsIndex, append(encodedTxsHashes, encodedScrsHashes...))
+	err = ei.removeIfHashesNotEmpty(elasticIndexer.LogsIndex, append(encodedTxsHashes, encodedScrsHashes...))
+	if err != nil {
+		return err
+	}
+
+	return ei.updateDelegatorsInCaseOfRevert(header, body)
+}
+
+func (ei *elasticProcessor) updateDelegatorsInCaseOfRevert(header coreData.HeaderHandler, body *block.Body) error {
+	// delegators index should be updated in case of revert only if the observer is in Metachain and the reverted block has miniblocks
+	shouldNotUpdate := header.GetShardID() != core.MetachainShardId || len(body.MiniBlocks) == 0
```

Comment:
faulty condition. based on the comment, only the combination `shardID == meta && len(miniblocks) > 0` should work, but with the current code, `shardID = 1 && len(miniblocks) = 1` will also work. refactor:
```
shouldUpdate := header.GetShardID() == core.MetachainShardId && len(body.MiniBlocks) > 0
if !shouldUpdate {
   return nil
}
```

Reply:
changed.